lib/methods/import.js

```diff
@@ -232,12 +236,55 @@ module.exports = self => {
       return locale;
     },
 
-    async insertDocs(req, docs, reporting) {
+    async checkDuplicates(req, docs) {
       const duplicatedDocs = [];
-      const duplicatedIds = [];
+
+      const docIds = docs
+        .filter(({ aposDocId }) => aposDocId)
+        .map(({ aposDocId }) => aposDocId);
+
+      const aposLocale = `${req.locale}:draft`;
+      const criteria = {
+        $and: [
+          { aposDocId: { $in: docIds } },
+          {
+            $or: [
+              { aposLocale: { $exists: false } },
+              { aposLocale }
+            ]
+          }
+        ]
+      };
+
+      const duplicates = await self.apos.doc.db.distinct('aposDocId', criteria);
+      const duplicatedIds = new Set(duplicates);
+
+      for (const doc of docs) {
+        if (duplicatedIds.has(doc.aposDocId) && (!doc.aposMode || doc.aposMode === 'draft')) {
+          duplicatedDocs.push({
+            aposDocId: doc.aposDocId,
+            title: doc.title,
+            type: doc.type,
+            updatedAt: doc.updatedAt
+          });
+        }
+      }
+
+      return {
+        duplicatedIds,
+        duplicatedDocs
```

Comment:
```suggestion
        // renamed
        duplicatedAposIds,
        
        // avoids for..of loop, easier to read I find
        duplicatedDocs: docs
          .filter(doc => doc.aposMode !== 'published')
          // could be in the filter above but easier to read that way
          .filter(doc => duplicatedIds.includes(doc.aposDocId))
          .map(doc => ({
            aposDocId: doc.aposDocId,
            title: doc.title,
            type: doc.type,
            updatedAt: doc.updatedAt
          });
```

optional

Reply:
I think we should prioritize performances for this module, might be a lot of documents, especially in a near future with `bigUpload`? (Like double filter should definitely be avoided)

Reply:
I changed this code but with one filter